pages/api/level/[id]/[slugName].ts

```diff
@@ -0,0 +1,45 @@
+import withAuth, { NextApiRequestWithAuth } from '../../../../lib/withAuth';
+import Level from '../../../../models/db/level';
+import { LevelModel } from '../../../../models/mongoose';
+import type { NextApiResponse } from 'next';
+import dbConnect from '../../../../lib/dbConnect';
+
+export default withAuth(async (req: NextApiRequestWithAuth, res: NextApiResponse) => {
```

Comment:
Shouldn't need withAuth for this API. We can model this after the existing level-by-id API (and maybe should call this level-by-username-and-slug)

Reply:
Oh right this would prevent guests from playing wouldnt it